internal/support/logging/interfaces/persistence.go

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+* Copyright 2019 Dell Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed under the License
+* is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+* or implied. See the License for the specific language governing permissions and limitations under
+* the License.
+*******************************************************************************/
+
+package interfaces
+
+import (
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+const (
+	PersistenceDB   = "database"
+	PersistenceFile = "file"
+)
+
+type Persistence interface {
+	Add(logEntry models.LogEntry) error
+	CloseSession()
+	Remove(criteria Criteria) (int, error)
+	Find(criteria Criteria) ([]models.LogEntry, error)
+
+	// Needed for the tests. Reset the instance (closing files, sessions...)
+	// and clear the logs.
+	Reset()
```

Comment:
Do we need to have this method in the interface, and/or is there a better way to achieve this? My concerns here are:  

1. We are now requiring implementations to provide functionality that is only useful during testing.
2. The perceived intention of this function is to purge all data to allow for a clean testing slate. An accidental call in production is more likely and unrecoverable, that is if the implementation is indeed purging data. For example, the Mongo implementation will permanently delete everything.